lxd/network/openvswitch/ovn.go

```diff
@@ -2047,3 +2082,154 @@ func (o *OVN) LogicalRouterPeeringDelete(opts OVNRouterPeering) error {
 
 	return nil
 }
+
+// GetHardwareAddress gets the mac address of the ovn managed network.
+func (o *OVN) GetHardwareAddress(chassisGroup OVNChassisGroup) (string, error) {
+	nameFilter := fmt.Sprintf("name=%s-lr-lrp-ext", chassisGroup)
+	data, err := o.nbctl("--data=bare", "--format=csv", "find", "Logical_Router_Port", nameFilter)
+	if err != nil {
+		return "", err
+	}
+
+	records := readCSV(data)
+	mac := ""
+	i := 0
+	for record := range records {
+		mac = record.values["mac"]
+		i++
+	}
+
+	if i != 1 || mac == "" {
+		return "", fmt.Errorf("Error getting hardware address")
+	}
+
+	return mac, nil
+}
+
+// GetMTU gets the mtu of the ovn managed network.
+func (o *OVN) GetMTU(chassisGroup OVNChassisGroup) (int, error) {
+	nameFilter := fmt.Sprintf("name=%s-lr-lrp-int", chassisGroup)
+	data, err := o.nbctl("--data=bare", "--format=csv", "find", "Logical_Router_Port", nameFilter)
+	if err != nil {
+		return 0, err
+	}
+
+	records := readCSV(data)
+	ipv6RAConfigs := ""
+	i := 0
+	for record := range records {
+		ipv6RAConfigs = record.values["ipv6_ra_configs"]
+		i++
+	}
+
+	if i != 1 || ipv6RAConfigs == "" {
+		return 0, fmt.Errorf("Error getting mtu")
+	}
+
+	kvPairs := strings.Fields(ipv6RAConfigs)
+	for _, kvPair := range kvPairs {
+		kv := strings.Split(kvPair, "=")
+		if len(kv) == 2 && kv[0] == "mtu" {
+			return strconv.Atoi(kv[1])
+		}
+	}
+
+	return 0, fmt.Errorf("Error getting mtu")
+}
+
+// GetChassis gets the name of the ovn chassis (host+gateway)
+func (o *OVN) GetChassis(chassisGroup OVNChassisGroup) (string, error) {
```

Comment:
This function should take a `OVNRouterPort` as an argument rather than `OVNChassisGroup` and let the `ovn` network driver generate the port name. You can then use the `ovn` network driver's `getRouterExtPortName` function to generate the port name.

There should be a comment added to the bit where `cr-` is prepended to the logical router port name, explaining why it is doing that (this is because it is looking for the associated "chassis redirect" port for the logical router port).